pkg/backend/snapshot.go

```diff
@@ -512,7 +512,8 @@ func (rsm *refreshSnapshotMutation) End(step deploy.Step, successful bool) error
 		// The exception to this is persisted refreshes, which are not elided and are treated as normal operations.
 		// These can either update or delete a resource.
 		refreshStep, isRefreshStep := step.(*deploy.RefreshStep)
-		if isRefreshStep && refreshStep.Persisted() {
+		viewStep, isViewStep := step.(*deploy.ViewStep)
+		if (isViewStep && viewStep.Persisted()) || (isRefreshStep && refreshStep.Persisted()) {
```

Comment:
Could (should) this be done as `persistableStep, isPersistableStep := step.(interface { Persisted() bool })`?